### PR TITLE
Currencies

### PR DESCRIPTION
**Pull request by ilmuwcql**

### CODE REVIEW

File: App.Common/Repositories/ICurrencyRepository.cs [hunk 1]
```diff
@@ -1,6 +1,11 @@
-﻿namespace App.Repositories
+﻿using System;
+using System.Collections.Generic;
+
+namespace App.Repositories
 {
     public interface ICurrencyRepository
     {
+        IEnumerable<string> GetCurrencyCodes();
+        IEnumerable<KeyValuePair<string, decimal>> GetExchangeRates(string code, DateTime date);
```
Comment (ilmuwcql) [inline App.Common/Repositories/ICurrencyRepository.cs]: Remove extra parameter code

File: App.Currencies/CurrencyManager.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public interface ICurrencyManager
+    {
+        IEnumerable<string> GetCurrencyCodes();
+        IEnumerable<KeyValuePair<string, decimal>> GetExchangeRate(string fromCode, DateTime date);
+    }
+
+    public class CurrencyManager : ICurrencyManager, ITransientDependency
+    {
+        private readonly ICurrencyRepository _repository;
+        private Dictionary<string, decimal> _exchangeRates;
```
Comment (ilmuwcql) [inline App.Currencies/CurrencyManager.cs]: Make it local variable in GetExchangeRate method

File: App.Currencies/Controllers/CurrenciesController.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            _logger = logger;
+            _currencyManager = currencyManager;
+        }
+
+        [HttpGet]
+        public ActionResult<IEnumerable<string>> GetCurrencyCodes()
+        {
+            var serviceCallResult = _currencyManager.GetCurrencyCodes()?.ToList();
+            if (serviceCallResult == null || serviceCallResult.Count == 0)
+                return NotFound(serviceCallResult);
```
Comment (ilmuwcql) [inline App.Currencies/Controllers/CurrenciesController.cs]: Remove extra if statement

File: App.Currencies/Controllers/CurrenciesController.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            return serviceCallResult;
+        }
+
```
Comment (ilmuwcql) [inline App.Currencies/Controllers/CurrenciesController.cs:40]: Add method GetCurrencyById

**Review by ghost — APPROVED** [review]

**Pull request merged**